packages/toml/src/index.ts

```diff
@@ -0,0 +1,66 @@
+import { Taplo } from '@taplo/lib'
+import type { Plugin } from 'prettier'
+
+import { defaultOptions, options } from './options'
+import type { PrettierOptions, TaploOptions } from './types'
+
+const PLUGIN_NAME = 'toml'
+
+let taplo: Taplo | null = null
+
+async function format(code: string, options: TaploOptions) {
+  if (!taplo) {
+    taplo = await Taplo.initialize()
+  }
+
+  return taplo.format(code, { options })
+}
+
+function removeBeginningTrailingNewline(code: string) {
```

Comment:
Is this required? Do you have any reference?

Reply:
Run this code:

```ts
import Taplo from '@taplo/lib'

console.log(
  (await Taplo.Taplo.initialize()).format(`



  a =1
  `)
)
```

Reply:
Is that an invalid toml already? That would be intended to not support invalid content.

Reply:
Ah sorry, use this:

```ts
import Taplo from '@taplo/lib'

console.log(
  (await Taplo.Taplo.initialize()).format(`



a =1
  `)
)

```

Output:

![image](https://github.com/un-ts/prettier/assets/58381667/8fba6fa4-4fed-4c8e-97ab-ccd553ee41aa)

Reply:
What's the problem here?

Reply:
And also, does an inline `.trim` do the same trick?

Reply:
> What's the problem here?

Not sure, this seemed to be the expected behavior of taplo.

> And also, does an inline .trim do the same trick?

I'm a noob 🤡